Sources/Algorithms/Partition.swift

```diff
@@ -204,3 +204,171 @@ extension Collection {
   }
 }
 
+//===----------------------------------------------------------------------===//
+// partitioned(_:)
+//===----------------------------------------------------------------------===//
+
+extension Sequence {
+  /// Returns two arrays containing, in order, the elements of the sequence that
+  /// do and don’t satisfy the given predicate, respectively.
+  ///
+  /// In this example, `partitioned(_:)` is used to separate the input based on
+  /// names that aren’t and are shorter than five characters, respectively:
+  ///
+  ///     let cast = ["Vivien", "Marlon", "Kim", "Karl"]
+  ///     let (longNames, shortNames) = cast.partitioned({ $0.count < 5 })
+  ///     print(longNames)
+  ///     // Prints "["Vivien", "Marlon"]"
+  ///     print(shortNames)
+  ///     // Prints "["Kim", "Karl"]"
+  ///
+  /// - Parameter belongsInSecondCollection: A closure that takes an element of
+  /// the sequence as its argument and returns a Boolean value indicating
+  /// whether the element should be included in the second returned array.
+  /// Otherwise, the element will appear in the first returned array.
+  ///
+  /// - Returns: Two arrays with with all of the elements of the receiver. The
+  /// first array contains all the elements that `belongsInSecondCollection`
+  /// didn’t allow, and the second array contains all the elements that
+  /// `belongsInSecondCollection` allowed.
+  ///
+  /// - Complexity: O(*n*), where *n* is the length of the sequence.
+  ///
+  /// - Note: This algorithm performs a bit slower than the same algorithm on
+  /// `RandomAccessCollection` since the size of the sequence is unknown, unlike
+  /// `RandomAccessCollection`.
+  @inlinable
+  public func partitioned(
+    _ belongsInSecondCollection: (Element) throws -> Bool
+  ) rethrows -> ([Element], [Element]) {
```

Comment:
I was thinking maybe `excluded` and `included`, respectively. While they would be accurate, they don’t really sound great alongside the `belongsInSecondCollection` parameter name. I think keeping the parameter name and behavior consistent across the partition* family of functions is important.

Reply:
We discussed this internally and we agree that the combination of `belongsInSecondCollection` in combination with `included`/`excluded` would be odd. However, note that the closure parameter of the mutating versions is called `belongsInSecondPartition` primarily because it's the only way to specify which side of the collection corresponds to the elements that produce `true`. Here we have tuple labels to work with though, which I think serves this purpose a lot better than the parameter name. How does that sound? `falseElements:` and `trueElements:` are the labels we came up with (in that order, to still stay as consistent as we can with the mutating versions). 

Let's also add an external `by:` label here, the mutating versions don't have one but there's no reason why they shouldn't. We'll probably want to change those later to include it as well.

Reply:
I think that’s probably the best thing we can do in this scenario. Thanks Tim and team!